src/recordpage.cpp

```diff
@@ -834,6 +858,46 @@ void RecordPage::setError(const QString& message)
     ui->applicationRecordErrorMessage->setVisible(!message.isEmpty());
 }
 
+void RecordPage::addKonsoleWidget()
+{
+    KService::Ptr service = KService::serviceByDesktopName(QLatin1String("konsolepart"));
+
+    if (!service) {
+        return;
+    }
+
+    const auto tail = QStandardPaths::findExecutable(QLatin1String("tail"));
+
+    if (tail.isEmpty()) {
+        return;
+    }
+
+    if (m_konsolePart) {
+        ui->recordOutputBox->layout()->removeWidget(m_konsolePart->widget());
+        m_konsolePart->deleteLater();
+    }
+    m_konsolePart = service->createInstance<KParts::ReadOnlyPart>(this);
+
+    if (!m_konsolePart) {
+        return;
+    }
+
+    m_konsoleFile = new QTemporaryFile(this);
+    m_konsoleFile->open();
+
+    const auto terminalInterface = qobject_cast<TerminalInterface*>(m_konsolePart);
+    if (!terminalInterface) {
+        qWarning("konsole kpart doesn't implement terminal interface");
+        delete m_konsolePart;
+        m_konsolePart = nullptr;
+        return;
+    }
+
+    terminalInterface->startProgram(tail, {QLatin1String("tail"), QLatin1String("-f"), m_konsoleFile->fileName()});
```

Comment:
use the `tail` here instead that you found above. and the `QLatin1String("-f")` should be a `QStringLiteral` instead